float32 distance
float32 relative_velocity
float32 angle
